src/vitabel/timeseries.py

```diff
@@ -1890,32 +1978,55 @@ def plot(
         if plotstyle is not None:
             base_plotstyle.update(plotstyle)
 
-        if data is not None:
-            # TODO: deal with data string entries (or disallow them)
-            if "fmt" not in base_plotstyle:
-                base_plotstyle["fmt"] = "none"
-            if "capsize" not in base_plotstyle:
-                base_plotstyle["capsize"] = 3
-            artist = plot_axes.errorbar(
-                time_midpoints, data, xerr=time_radius, **base_plotstyle
-            )
-        else:
+        if plot_type in {'box', 'combined'}:
             if "alpha" not in base_plotstyle:
-                base_plotstyle["alpha"] = 0.2
+                base_plotstyle.update({"alpha": 0.2})
             if "color" not in base_plotstyle:
-                base_plotstyle["color"] = "blue"
+                base_plotstyle.update({"color" :"blue"})
             if not any(k in base_plotstyle for k in ('facecolor', 'fc')):
-                base_plotstyle["facecolor"] = base_plotstyle["color"]
-                
-            for xmin, xmax in time_index:
+                base_plotstyle.update({"facecolor" : base_plotstyle["color"]})
+            
+            if plot_type == 'combined'and data is not None:
+                box_time_index = time_index[np.isnan(data)]
+            else:
+                box_time_index = time_index
+           
+            for i, (xmin, xmax) in enumerate(box_time_index):
                 # Filter for props in kwargs which can be handled by axvspan / pathces.Rectangle
                 rectangle_props = Rectangle((0, 0), 1, 1).properties().keys()
                 filtered_base_plotstyle = {k: v for k, v in base_plotstyle.items() if k in rectangle_props}
                 artist = plot_axes.axvspan(xmin, xmax, **filtered_base_plotstyle)
+                if i == 0:  # add a legend only for the first rectangle
+                    artist.set_label(self.name)
+
+        if plot_type == "hline" and len(self) > 0 and data is None:
+            logger.warning(
+                f"Label {self.name} has no data, skipping horizontal line plot"
+            )
+            return figure
         
-        # check if any artist was generated
-        if artist is not None:
-            artist.set_label(self.name)  # only set legend once
+        if plot_type in {'hline', 'combined'} and data is not None:
+            
+            if plot_type == 'combined':
+                hline_time_index = time_index[~np.isnan(data)]
+                hline_data = data[~np.isnan(data)]  
+            else:
+                hline_time_index = time_index 
+            
+            time_midpoints = np.mean(hline_time_index, axis=1)
+            time_radius = np.diff(hline_time_index, axis=1).reshape(-1) / 2.0
+
+            # TODO: deal with data string entries (or disallow them)
```

Comment:
disallowing isn't really a choice any more. an issue would be better than this comment.

```suggestion
            # TODO: deal with text_data
```